src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -926,16 +1020,7 @@ def main():
         )
 
     finally:
-        # stop callback server if it is on
-        if conf:
-            conf.stop_callback_server()
-
-        # stop podman service if it is on
-        if proc:
-            proc.terminate()
-
-        if not keep_job_dir:
-            shutil.rmtree(job_dir)
+        _cleanup_resources(conf, proc, job_dir, keep_job_dir)
```

Comment:
Even with these changes main is still > 100 LOC, which is way too long

Reply:
Although maybe we should merge this and then take pass 2 to keep the PR small

Reply:
I actually already did further reduce the main().

Reply:
Now it's 60 lines, which is still on the large side but manageable

Reply:
:+1: